.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ 'main' ]
+  pull_request:
+    branches: [ 'main' ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js 
+      uses: actions/setup-node@v3.5.0
+      with:
+        node-version: '16.x'
+        cache: 'npm'
+    - name: Install dependencies
+      run: npm install
+      
+    - name: echo installed is done
+      run: echo Installed npm packages ...
+      
+    - name: Run the tests
+      run: npm test
```

Comment:
Instead of running a unit tests here (which are not configured yet) for now, we can run this `npm run lint --max-warnings=0` which will make `eslint` go through the code and will fail if there are any `warnings` and `errors` which are not handled. @ashavijit what do you think 🤔 ?

Reply:
This is good. Jest is the professional way of handling error! For now as the unit tests are not configured yet. It can implemented temporarily.

Reply:
@manojkrishnak are you working on it . If not I will assign someone or will do it myself

Reply:
can this be done with basic test just to check the test's phase is working or not. Or do you want a detailed tests ?

Reply:
if detailed one,it will be good!

Reply:
if you dont see anything from me by tonight, you can take this up.